```
 0:00.11 INFO Installing browser
Traceback (most recent call last):
  File "./wpt", line 5, in <module>
    wpt.main()
  File "/home/kriti/mozdev/web-platform-tests/tools/wpt/wpt.py", line 132, in main
    rv = script(*args, **kwargs)
  File "/home/kriti/mozdev/web-platform-tests/tools/wpt/run.py", line 404, in run
    **kwargs)
  File "/home/kriti/mozdev/web-platform-tests/tools/wpt/run.py", line 382, in setup_wptrunner
    kwargs["binary"] = setup_cls.install(venv)
  File "/home/kriti/mozdev/web-platform-tests/tools/wpt/run.py", line 335, in install
    raise NotImplementedError
NotImplementedError
```

--- a/tools/wpt/browser.py
+++ b/tools/wpt/browser.py
@@ -431,8 +431,37 @@ class Servo(Browser):
     product = "servo"
     requirements = "requirements_servo.txt"
 
+    def platform_components(self):
+        platform = {
+            "Linux": "linux",
+            "Windows": "win",
+            "Darwin": "mac"
+        }.get(uname[0])
+
+        if platform is None:
+            raise ValueError("Unable to construct a valid Servo package for current platform")
+
+        if platform == "linux":
+            extension = ".tar.gz"
+            decompress = untar
+        elif platform == "win" or platform == "mac":
+            raise ValueError("Unable to construct a valid Servo package for current platform")
+
+        return (platform, extension, decompress)
+
     def install(self, dest=None):
-        raise NotImplementedError
+        """Install latest Browser Engine."""
+        if dest is None:
+            dest = os.pwd
+
+        platform, extension, decompress = self.platform_components()
+        url = "https://download.servo.org/nightly/%s/servo-latest%s" % (platform, extension)
+
+        decompress(get(url).raw, dest=dest)
+        path = find_executable("servo", os.path.join(dest, "servo"))
+        st = os.stat(path)
+        os.chmod(path, st.st_mode | stat.S_IEXEC)
+        return path
 
     def find_binary(self, path=None):
         return find_executable("servo")
@@ -444,7 +473,9 @@ class Servo(Browser):
         raise NotImplementedError
 
     def version(self, root):
-        return None
+        """Retrieve the release version of the installed browser."""
+        output = call(self.binary, "--version")
+        return re.search(r"[0-9\.]+( [a-z]+)?$", output.strip()).group(0)
 
 
 class Sauce(Browser):

--- a/tools/wpt/run.py
+++ b/tools/wpt/run.py
@@ -332,7 +332,8 @@ class Servo(BrowserSetup):
     browser_cls = browser.Servo
 
     def install(self, venv):
-        raise NotImplementedError
+        if self.prompt_install(self.name):
+            return self.browser.install(venv.path)
 
     def setup_kwargs(self, kwargs):
         if kwargs["binary"] is None:

